run_xnat2bids.py

```diff
@@ -6,52 +7,117 @@
+                        if value == True: x2b_param_list.append(arg) 
+                    # Extract parameters from include / skip lists
+                    elif param in param_lists:
+                        arg = extractParams(param, value)
+                        x2b_param_list.append(arg)
+                    # Other arguments follow --param value format.
+                    else:
+                        arg = f"--{param} {value}"
+                        x2b_param_list.append(arg)
+
+        elif section_name == "slurm-args":
+            for param, value in section_dict.items():
+                if value != "" and value is not None:
+                    arg = f"--{param} {value}"
+                    slurm_param_list.append(arg)
+
+    return x2b_param_list, slurm_param_list
+
+def main():
+    # Instantiate argument parserß
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", help="path to user config")
+    args = parser.parse_args()
+
+    # Load default config file into dictionary
+    default_params = load('/gpfs/data/bnc/scripts/x2b_default_config.toml')
+
+    # Set arglist. If user provides config, merge dictionaries.
+    if args.config is None:
+        arglist = default_params
+    else:
+        user_params = load(args.config)
+        default_params.update(user_params)
+        arglist = user_params
+
+    # Fetch user credentials 
+    user = input('Enter Username: ')
+    password = getpass('Enter Password: ')
+
+    # Fetch compiled xnat2bids abd slurm parameter lists
+    x2b_param_list, slurm_param_list, user = compileArgumentList(arglist, user)
+
+    # Insert username and password into x2b_param_list
+    x2b_param_list.insert(2, f"--user {user}")
+    x2b_param_list.insert(3, f"--pass {password}")
+
+    # Fetch latest version if not provided
+    if(arglist['xnat2bids-args']['version'] == ""):
+        list_of_versions = glob.glob('/gpfs/data/bnc/simgs/brownbnc/*') 
+        latest_version = max(list_of_versions, key=os.path.getctime)
+        version = latest_version.split('-')[-1].replace('.sif', '')
+
+    # Define singularity image 
+    simg=f"/gpfs/data/bnc/simgs/brownbnc/xnat-tools-{version}.sif"
+
+    output = arglist['slurm-args']['output']
+    if not (os.path.exists(os.path.dirname(output))):
+        os.mkdir(os.path.dirname(output))
+
+    # Display formatted command to user
+    new_line = '\n'
+    print(f"\033[1mRunning xnat2bids:\033[0m\n"
+      f"\033[1m----------------------------------------\033[0m\n"
+      f"\033[1msrun\033[0m      \n{new_line.join(slurm_param_list)} \n\n"
+      f"\033[1msingularity exec --contain -B /gpfs/data/bnc {simg}\033[0m \n\n"
+      f"\033[1mxnat2bids\033[0m     \n{new_line.join(x2b_param_list)}"
+      f"\033[1m----------------------------------------\033[0m")
+
+    # Run xnat2bids via srun
+    subprocess.run(shlex.split(
+        f"srun {' '.join(slurm_param_list)} \
+        singularity exec --contain -B /gpfs/data/bnc {simg} \
```

Comment:
We don't need to bind `/gpfs/data/bnc`, we need to bind directories that we need to write or read from. These include where the output will go and if the user passes a `bidsmap` file, then we need access to. 

I also the `--contain` or the `--no-home` means home is not shared and you may also need to bind `scratch`?